File: app/src/main/java/com/icostel/arhitecturesample/api/session/SessionData.java
package com.icostel.arhitecturesample.api.session;

import javax.inject.Inject;
import javax.inject.Singleton;

@Singleton
class SessionData {
    // token received after user logins with user & pass, used for making API calls
    String userToken = null;

    @Inject
    public SessionData() {

    }

    String getUserToken() {
        return userToken;
    }

    void setUserToken(String userToken) {
        this.userToken = userToken;
    }
}

File: app/src/main/java/com/icostel/arhitecturesample/api/session/SessionData.kt
package com.icostel.arhitecturesample.api.session

import javax.inject.Singleton

@Singleton
data class SessionData
constructor(
        internal var userToken: String? = null,
        internal var keepLogIn: Boolean = false,
        internal var userEmail: String? = null,
        internal var userPassword: String? = null
)
